### PR TITLE
Class

### PR DESCRIPTION
**Pull request by asmodehn**
pitch there : https://gitpitch.com/pyros-dev/ros1_template/class?grs=github

### CODE REVIEW

File: PITCHME.md [hunk 1]
```diff
@@ -0,0 +1,530 @@
...
+
+ROS1 Nodes
+----------
+
+`rospy.init_node()` or `ros::init()`
+- declares the current process as a ROS node
+- <span style="color:green">GOOD</span> : one node match one process
+- <span style="color:red">BAD</span> : special `rosmaster`, `rosout` nodes
+- <span style="color:red">BAD</span> : pollute your process global state
+- <span style="color:red">BAD</span> : async programming can be VERY confusing
```
Comment (AlexReimann) [inline PITCHME.md:141]: Maybe mention multi-threaded spinner?

File: PITCHME.md [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+C++ has nodelets, which means that one process has N node(let)s
```
Comment (AlexReimann) [inline PITCHME.md:142]: Should make extra slide for nodelet?

File: PITCHME.md [hunk 3]
```diff
@@ -0,0 +1,530 @@
...
+`rospy.Publisher()` or `ros::Publisher`
+
+`rospy.Subscriber()` or `ros::Subscriber`
+
+- allow ROSTCP/ROSUDP Push communication / event notifications
+- <span style="color:green">GOOD</span> : low overhead (binary serialization)
+- <span style="color:red">BAD</span> : lossy
+- <span style="color:red">BAD</span> : Inverted control flow (deal with your callback threadS)
+- <span style="color:red">BAD</span> : Cannot be used to interface with non-ROS systems
+- <span style="color:green">GOOD</span> : need 2 to implement pull communication
```
Comment (AlexReimann) [inline PITCHME.md:194]: Good?

File: PITCHME.md [hunk 4]
```diff
@@ -0,0 +1,530 @@
...
+Mutate client into a node
+-------------------------
+
+- Can we improve the communication
+- Is it really better ?
+- What are the problems or risks ? 
+
++++
+
+Review
```
Comment (AlexReimann) [inline PITCHME.md:378]: That's the next class already?

File: PITCHME.md (hunk 1 above)
Comment (asmodehn) [reply PITCHME.md:141]: Not sure what you mean or maybe I just dont know what it is...

File: PITCHME.md (hunk 2 above)
Comment (asmodehn) [reply PITCHME.md:142]: Maybe if you think it s worth it...

File: PITCHME.md (hunk 4 above)
Comment (asmodehn) [reply PITCHME.md:378]: Yeah, only one presentation for both.
Mmaybe I should add a "break" slide before this one ?

File: PITCHME.md (hunk 3 above)
Comment (asmodehn) [reply PITCHME.md:194]: I was thinking about communication patterns: implementing a service style communication with topic is doable and can simplify your system (by using only one communication implementation underneath)  : http://wiki.ros.org/rocon_python_comms/Tutorials/indigo/What%20are%20Service%20Pairs%3F

**Pull request merged**